### PR TITLE
Navigate to Source YAML editor when imports fail

### PR DESCRIPTION
**Pull request by ericpgreen2**
This PR:
-  Modifies the "add source" actions so that, when an import fails, the user is navigated to the Source YAML Editor w/ an error in the console
- Returns the "Create model" CTA to the header in the Source workspace

[Notion PRD](https://www.notion.so/rilldata/Adapt-add-source-modal-to-generate-source-config-when-errors-are-present-8f8e5738b65b4cc39ea475345b86f498).

This PR follows #2770.

### CODE REVIEW

File: web-common/src/features/sources/modal/LocalSourceUpload.svelte [hunk 1]
```diff
@@ -134,6 +114,9 @@
           filePath
         );
       }
+
+      // Navigate to source page
+      goto(`/source/${tableName}`);
```
Comment (AdityaHegde) [inline web-common/src/features/sources/modal/LocalSourceUpload.svelte:119]: Having an issue here would block user from reuploading another file right? Is that the UX we are going for?

File: web-common/src/features/sources/workspace/SourceWorkspaceHeader.svelte [hunk 2]
```diff
@@ -166,18 +172,37 @@
   );
   $: isSourceUnsaved = $isSourceUnsavedQuery.data;
 
-  $: reconciliationErrors = getFileArtifactReconciliationErrors(
-    $fileArtifactsStore,
-    `${sourceName}.yaml`
-  );
+  const handleCreateModelFromSource = async () => {
+    const modelName = await createModelFromSourceV2(queryClient, sourceName);
+    goto(`/model/${modelName}`);
```
Comment (AdityaHegde) [inline web-common/src/features/sources/workspace/SourceWorkspaceHeader.svelte:177]: Do we not need to redirect to modeller if there is an error in creating the model?

File: web-common/src/layout/workspace/WorkspaceHeader.svelte [hunk 3]
```diff
@@ -51,6 +51,7 @@
     Math.max((editingTitle ? titleInputValue : titleInput)?.length || 0, 5) + 1;
 
   $: width = $observedNode?.getBoundingClientRect()?.width;
+  $: console.log("width", width);
```
Comment (AdityaHegde) [inline web-common/src/layout/workspace/WorkspaceHeader.svelte:54]: Remove this console.log

File: web-common/src/features/sources/modal/RemoteSourceForm.svelte [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+      createForm({
+        initialValues: {
+          sourceName: "", // avoids `values.sourceName` warning
+        },
+        validationSchema: yupSchema,
+        onSubmit: async (values) => {
+          overlay.set({ title: `Importing ${values.sourceName}` });
+          try {
+            await submitRemoteSourceForm(queryClient, connector.name, values);
+            goto(`/source/${values.sourceName}`);
```
[the remark this answers is not in the document]
Comment (ericpgreen2) [reply web-common/src/features/sources/modal/RemoteSourceForm.svelte:55]: If `submitRemoteSourceForm` isn't able to successfully Put and Reconcile, the `/source/{sourceName}` route would throw a 404

File: web-common/src/features/sources/modal/LocalSourceUpload.svelte (hunk 1 above)
Comment (ericpgreen2) [reply web-common/src/features/sources/modal/LocalSourceUpload.svelte:119]: Well, a couple things:
- For this PR, the desired UX is to send the user to the Source YAML editor on error
- In a future PR, we need to make distinct actions for "upload file" (add to `data/`) and "create source" (create the Source YAML file)

File: web-common/src/features/sources/workspace/SourceWorkspaceHeader.svelte (hunk 2 above)
Comment (ericpgreen2) [reply web-common/src/features/sources/workspace/SourceWorkspaceHeader.svelte:177]: What are you suggesting? If `createModelFromSourceV2` throws an error, the `goto` won't get hit.

File: web-common/src/layout/workspace/WorkspaceHeader.svelte (hunk 3 above)
Comment (ericpgreen2) [reply web-common/src/layout/workspace/WorkspaceHeader.svelte]: Done

**Review by AdityaHegde — APPROVED** [review]

File: web-common/src/features/sources/modal/RemoteSourceForm.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/features/sources/modal/RemoteSourceForm.svelte:55]: I was a little thrown off by the fact that reconcile doesnt fail the network call but a response is returned.

File: web-common/src/features/sources/workspace/SourceWorkspaceHeader.svelte (hunk 2 above)
Comment (AdityaHegde) [reply web-common/src/features/sources/workspace/SourceWorkspaceHeader.svelte:177]: Ya that is what i was thinking. But this should probably be handled when we integrate the new reconcile. Any `put` success should navigate to the page, reconcile errors will be filled asynchronously by the new reconcile.

**Pull request merged**